ui/src/app/machines/components/ActionFormWrapper/CloneForm/CloneForm.tsx

```diff
@@ -0,0 +1,68 @@
+import { useSelector } from "react-redux";
+import * as Yup from "yup";
+
+import CloneFormFields from "./CloneFormFields";
+
+import ActionForm from "app/base/components/ActionForm";
+import type { ClearSelectedAction } from "app/base/types";
+import { actions as machineActions } from "app/store/machine";
+import machineSelectors from "app/store/machine/selectors";
+import type { Machine } from "app/store/machine/types";
+import { NodeActions } from "app/store/types/node";
+
+type Props = {
+  actionDisabled?: boolean;
+  clearSelectedAction: ClearSelectedAction;
+};
+
+export type CloneFormValues = {
+  interfaces: boolean;
+  source: Machine["system_id"];
+  storage: boolean;
+};
+
+const CloneFormSchema = Yup.object()
+  .shape({
+    interfaces: Yup.boolean(),
+    source: Yup.string().required("Source machine must be selected."),
+    storage: Yup.boolean(),
+  })
+  .defined();
+
+export const CloneForm = ({
+  actionDisabled,
+  clearSelectedAction,
+}: Props): JSX.Element => {
+  const machines = useSelector(machineSelectors.all);
```

Comment:
For machine details we'll need fetch all the machines if they haven't been loaded already (not suggesting you need to do that in this branch).

Reply:
Thanks, yeah hadn't considered that yet